### PR TITLE
Add time series DSL module

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Design issue

**What is the current behavior?** *(You can also link to an open issue here)*
`powsybl-time-series-api` depends on Groovy library and consequently `powsybl-dynamic-simulation-api` too.
Groovy dependency is not always necessary and sometimes could cause some issue like for GraalVM native image compilation. 

**What is the new behavior (if this is a feature change)?**
Part of `powsybl-time-series-api` code regarding Groovy DSL has be moved to new `powsybl-time-series-dsl` module.  

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/CalculatedTimeSeriesDslLoader.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.ServiceLoader;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public interface CalculatedTimeSeriesDslLoader {
+
+    Map<String, NodeCalc> load(String script, ReadOnlyTimeSeriesStore store);
+
+    static CalculatedTimeSeriesDslLoader find() {
```
Comment (miovd) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/CalculatedTimeSeriesDslLoader.java:22]: Should we have have a "name" identifier for dsl loaders? Or we take one randomly? Please note that it may be done in another PR.

**Review by miovd — CHANGES_REQUESTED** [review]
Don't forget to add time-series-dsl in distribution pom

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/CalculatedTimeSeriesDslLoader.java (hunk 1 above)
Comment (geofjamg) [reply time-series/time-series-api/src/main/java/com/powsybl/timeseries/CalculatedTimeSeriesDslLoader.java:22]: As there is no plan for adding a second DSL language and worst no plan for dealing with 2 different DSL scripts engine at the same time in the classpath I would stay with a simple impl that suppose there is only one in the classpath.
So you agree with this proposition, my code is not correct as we should also check there is only one and I will add a check to ensure unicity.

**Review by miovd — APPROVED** [review]
